lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -24,6 +44,47 @@ cudaDeviceInterface::~cudaDeviceInterface() {
     cleanup_memory();
 }
 
+void cudaDeviceInterface::set_device_clocks(unsigned int mem_clock, unsigned int core_clock,
+                                            unsigned int mem_count, unsigned int core_count) {
+
+    // Set default clocks to zero
+    uint32_t gpu_mem_clock = config.get_default<uint32_t>(unique_name, "gpu_mem_clock", 0);
+    uint32_t gpu_core_clock = config.get_default<uint32_t>(unique_name, "gpu_core_clock", 0);
+
+    uint32_t get_gpu_mem_clock, get_gpu_core_clock;
+
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_mem_clock, get_gpu_mem_clock);
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_core_clock, get_gpu_core_clock);
```

Comment:
I don't think these are being called correctly.  The second argument should be `nvmlClockType_t`, e.g. `NVML_CLOCK_MEM`.   However, I don't think you need to call these at all.  Since the list of clocks from `nvmlDeviceGetSupported*Clocks` will have the max clock it it.

Reply:
My initial intent was to get the maximum GPU clock in formation and, check it against the current set device clock, thank you for the information regarding  `nvmlDeviceGetSupported*Clocks`. That solves the query.